### PR TITLE
feat: support insecure access to the nodes running in maintenance

### PR DESCRIPTION
**Pull request by Unix4ever**
Any insecure `talosctl` commands now work with Omni per-instance `talosconfig`.
User should have at least `Operator` Omni role to be able to use the insecure `talosctl` mode.

DNS resolver was updated to react on the `MachineStatus` resource creation, not only the `ClusterMachineConfigStatus` resource. That makes the DNS record for UUID appear as soon as machine joins Omni, not when the machine gets allocated into a cluster.

Machines list now has maintenance Talos version update button. The UI will issue `talosctl upgrade` when another Talos version is picked.

`MachineStatus` controller was updated a bit: version poller wasn't marked as dirty after maintenance upgrades. Now we mark it as dirty every time we get Talos `MachineStatus` resource update.

Also fixed UI issues here and there.

Fixes: #284 
Fixes: #236

### CODE REVIEW

File: frontend/src/views/omni/ItemLabels/ItemLabels.vue [hunk 1]
```diff
@@ -90,7 +87,15 @@ const labels = computed((): Array<Label> => {
   }
 
   labelsArray.sort((a, b) => {
-    return getLabelOrder(a) - getLabelOrder(b);
+    if (getLabelOrder(a) === getLabelOrder(b)) {
```
Comment (Unix4ever) [inline frontend/src/views/omni/ItemLabels/ItemLabels.vue:90]: Label order was broken.

File: frontend/src/views/omni/Overview/Overview.vue [hunk 2]
```diff
@@ -278,7 +278,7 @@ url: "${talosLoggingKernel}"
 }
 
 .overview-copy-icon {
-  @apply cursor-pointer w-5 text-primary-P2 p-0.5 rounded;
+  @apply cursor-pointer w-5 text-primary-P2 p-0.5 rounded w-5 h-5;
```
Comment (Unix4ever) [inline frontend/src/views/omni/Overview/Overview.vue:281]: fixes icons gigantomania

**Comment by Unix4ever** [comment]
https://github.com/siderolabs/omni/assets/830532/a6915a9f-aab6-495d-8a92-eb2e33b7e16d

File: frontend/src/methods/key.ts [hunk 3]
```diff
@@ -24,7 +24,6 @@ import {
   workloadProxyPublicKeyIdCookie,
   workloadProxyPublicKeyIdSignatureBase64Cookie,
 } from '@/api/resources';
-// import { Buffer } from "buffer";
```
Comment (utkuozdemir) [inline frontend/src/methods/key.ts:27]: maybe remove?

Comment (Unix4ever) [reply frontend/src/methods/key.ts:27]: it was removed 🤔

Comment (utkuozdemir) [reply frontend/src/methods/key.ts:27]: oh sorry, I'm blind, it's red :D

File: internal/backend/grpc/router/router.go [hunk 4]
```diff
@@ -154,16 +156,22 @@ func (r *Router) Director(ctx context.Context, fullMethodName string) (proxy.Mod
...
 	clusterName := getClusterName(md)
 
-	if backend, ok := r.talosBackends.Get(clusterName); ok {
+	id := clusterName
+
+	if clusterName == "" {
+		id = getNodeID(md)
+	}
+
+	if backend, ok := r.talosBackends.Get(id); ok {
```
Comment (Unix4ever) [inline internal/backend/grpc/router/router.go:168]: use struct as a key

Comment (Unix4ever) [reply internal/backend/grpc/router/router.go:168]: or add prefix to `cluster` and `machine` backends if we stick to strings.

**Comment by Unix4ever** [comment]
verify insecure access in the integration tests

**Pull request merged**